### PR TITLE
obs-ffmpeg: Fix memory leak with mpegts

### PR DESCRIPTION
**Pull request by pkviet**
### Description
Fixes https://github.com/obsproject/obs-studio/issues/7305.
Some copied data was not freed and also not all paths freed AVPackets.

### Motivation and Context
Leaks are bad.

### How Has This Been Tested?
Tested the leak was gone when streaming at very high bitrate with srt or rist.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
Use [special keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) to automatically link this to an issue. See also:
* https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

For example:
Fixes #7305

File: plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c [hunk 1]
```diff
@@ -1115,6 +1114,8 @@ void mpegts_write_packet(struct ffmpeg_output *stream,
 
 	pthread_mutex_lock(&stream->write_mutex);
 	da_push_back(stream->packets, &packet);
+	av_free(packet->data);
+	packet = NULL;
 	pthread_mutex_unlock(&stream->write_mutex);
```
Comment (notr1ch) [inline plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:1119]: Is this safe? This pushes a pointer to the packet into `stream->packets` and then immediately frees the data pointer, leaving `packet->data` pointing to freed memory. Did you mean to set `packet->data = NULL` or use `av_freep`?

Comment (pkviet) [reply plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:1122]: av_freep indeed; i moved it after av_interleaved_write_frame too.

Comment (notr1ch) [reply plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c:1121]: `packet = NULL;` now seems redundant as it's a stack variable that's no longer referenced on this code path.

Comment (pkviet) [reply plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c]: indeed, thx a lot as usual @notr1ch !

**Pull request merged**